library/Slingshot.php

```diff
@@ -251,7 +251,14 @@ private function processDocumentsMigration()
     {
         $searchHash = $this->migrationHash['from'];
         if ($this->searchQueryHash) {
+          if (!isset($this->searchQueryHash['query']))
```

Comment:
I would have deal with that in the intial setter of `$this->searchQueryHash`

```PHP
public function migrate(array $searchQueryHash = [], $convertDocCallBack = null)
{
    ...
    $this->searchQueryHash = isset($searchQueryHash['query']) ? $searchQueryHash : ['query' => $searchQueryHash];
    ...
}
```

then 
```PHP
private function processDocumentsMigration()
{
    ...
    $searchHash['body'] = $this->searchQueryHash;
    ...

```